rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,109 @@
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.gnosis.modules.sdai.constants import (
+    CPT_SDAI,
+    GNOSIS_SDAI_ADDRESS,
+    SDAI_ICON,
+    SDAI_LABEL,
+)
+from rotkehlchen.constants.assets import A_SDAI
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.gnosis.node_inquirer import GnosisInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+SDAI_DEPOSIT = b'\xdc\xbc\x1c\x05$\x0f1\xff:\xd0g\xef\x1e\xe3\\\xe4\x99wbu.:\tR\x84uED\xf4\xc7\t\xd7'  # noqa: E501
+SDAI_REDEEM = b'\xfb\xdey} \x1ch\x1b\x91\x05e)\x11\x9e\x0b\x02@|{\xb9jJ,u\xc0\x1f\xc9fr2\xc8\xdb'
+
+
+class SdaiDecoder(DecoderInterface):
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.sdai = A_SDAI.resolve_to_evm_token()
+
+    def _decode_sdai_deposit_events(self, context: DecoderContext) -> DecodingOutput:
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(amount_raw, None)
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:])
+        shares = token_normalized_value_decimals(shares_raw, None)
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                asset = event.asset.resolve_to_crypto_asset()
+                event.counterparty = CPT_SDAI
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {asset.name} to SDAI contract'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == shares
+            ):
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.balance.amount} SDAI'
+                event.counterparty = CPT_SDAI
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_sdai_redeem_events(self, context: DecoderContext) -> DecodingOutput:
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:])
```

Comment:
same, be explicit with the range

```suggestion
        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:64])
```